### PR TITLE
Rebalance dedicated melee weapons

### PR DESCRIPTION
**Pull request by Coolthulhu**
- Sledgehammer gets updated description (that it's never good), damage buff and weight reduction
- War hammer gets 33% buff in damage, becoming an actually good weapon. Not katana grade, but at 37 damage per 100 moves it is in "weapons of war" tier
- Mace and morningstar get small buff to damage and move from "OK, I guess" to "this is a weapon. At 34 and 33 "dp100" (damage per 100 moves) respectively
- Diamond machete gets a damage buff. It is still one of the shittiest diamond weapons at mere 40 dp100
- Diamond katana gets a giant nerf and moves from godlike tier to "still the best melee weapon" tier at 48 dp100
- Wakizashi gets a nerf: 35 dp100
- Katana gets a huge nerf and moves into a tier occupied by nodachi, rapier, broadsword and (!)diamond zwei. 40 dp100
- Firekatana gets the same nerf as katana, but also weight and volume reduction. It is still a downgrade to katana at 37 dp100
- Zwei and firezwei get small buff to damage. Both are still barely in "weapons of war" at 34 dp100
- Fire axe gets a minor buff that makes it 29 dp100
- Battle axe gets a GIANT buff that moves it from "below the scale, why would you use that over your fists" to "weapons of war". 36 dp100
- Survivor machete gets a moderate nerf. It's just a machete with a bunch of duct tape, it deserves less than what it has. 36 dp100
- Diamond zwei gets the small zwei buff and then some. Worst "big" diamond weapon at 40 dp100
- Diamond nodachi gets a moderate nerf. 45 dp100

How did I decide what to nerf/buff:
- The most important was obviously damage per moves. While very low damage per very low time could look much more useful on paper than in game (due to semi-rare armored enemies), no weapon should be the master of all trades. Looking at you, katana.
- RAPID technique was assumed to be used all the time where available. No other techniques considered. Most of them are either too situational (AoE, brutal strike), too unimportant (block) or didn't appear on the powerful weapons (reach)
- A tiny bit of realism. Battle axe is an axe made for battle, not a giant double blade monstrosity for barbarian cosplay.
- Tried to get diamond weapons close to 45 dp100 and "weapons of war" to 35-40 dp100 range
- Assumed 5 in all skills. At lower skills, cutting weapons would be better. At higher levels bashing ones get devastating crits, but this requires good accuracy.
- Assumed 0 dexterity and low encumbrance. Good dexterity would help with fast weapons more, encumbrance would hurt fast weapons more, but most of the weapons keep their tiers at dexterity 20 and at move penalty to encumbrance around 40
- Assumed "unlimited" strength (ie. enough to get full bashing damage). This mace/morningstar buff doesn't actually make them good, just makes it possible for them to be good.
- Assumed no stuck penalty. Katanas don't get stuck, broadswords, rapiers and war hammers might.
- Didn't consider Niten (nor any other MAs). Otherwise I'd nerf that katana even more, because kataniten is still nearly a total no-brainer if you can get both of them on one dude.
- Out of the powerful weapons, broadsword and rapier both have +1 to_hit compared to katanas, battle axes etc. They may deserve a nerf because of that - I wasn't sure so I didn't give it to them, though.

Trivia:
- Highest dp100 is now combat chainsaw (on) at 48, but it has -4 to_hit making it quite useless
- Fire weapons suck. No reason to nerf own sword with that extra weight and volume.
- Chainsaw weapons suck. WH40K can't stop here, this is weeb country
- Vacuum packed meat slice makes a comparable weapon to a hammer. Eat this!

### CODE REVIEW

**Comment by vache** [comment]
I have no specific comments on the suggested changes other than I'm glad they're getting looked at, but I would like to suggest making the exact calculation formula, the expected dpt ranges for each weapon tier, and some criteria for determining what tier a weapon should belong to available in some documentation or "adding a new weapon?" guide.  Hopefully, having objective calculations like that available would reduce the need to rebalance like this on this scale in the future.

**Comment by kevingranade** [comment]
Seconded, even a copy/paste of your PR summary in GAME_BALANCE.md would be better than nothing.

**Comment by chaosvolt** [comment]
Ah, that reminds me. I should see how best to rebalance the Medieval mod weapons relative to these, along with the few weapons in Arcana based off existing weaponry.

By the way, what about the awl pike? Last I checked, it was too slow to be practical.

**Comment by Coolthulhu** [comment]
Added the balancing notes to GAME_BALANCE.md. A bit more complex than most notes there, but should still be readable.

File: doc/GAME_BALANCE.md [hunk 1]
```diff
@@ -97,6 +97,48 @@ A measure of how well-suited the item is for being swung/thrust/etc. This factor
...
+### Weapon tiers
+Damage per turn should put the weapon into one of those categories:
+
+<10 - Not weapons. Those items may be pressed into service, but are unlikely to be better than fists. Plastic bottles, rocks, boots.
+
+11-15 - Tools not meant to strike and improvised weapons. Two-by-fours, pointy sticks, pipes, hammers.
+
+16-25 - Dangerous tools or weak dedicated weapons. Golf clubs, two-by-swords, wooden spears, knife spears, hatchets, switchblades, tonfas, quarterstaves.
+
+26-35 - The most dangerous of tools that aren't made to be weapons. Wood and fire axes, steel spears, electric carvers, kukris, bokken, machetes, barbed wire bats.
```
Comment (vache) [inline doc/GAME_BALANCE.md]: Nitpicking here, but I'd make this description include "improvised weapons", as a barbed wire bat is clearly made to be a weapon.  Could get even more specific and make the distinction between "crudely weapons, improvised weapons, professionally made weapons".

**Comment by Coolthulhu** [comment]
Does this sound better?
I tried to keep it in one line

**Comment by md000** [comment]
Maybe for another PR, but I believe parry vs block should also be looked into. One example that instantly springs to mind would be the broadsword, which with it's crossguard should be perfectly suited for blocking, probably moreso than the rapier (flimsy blade), forged sword (kinda short) or katana (disc guard). I would suggest giving block only to weapons that either have protrusions suitable for catching a weapon or are a solid piece of hard material possible to easily hold in a way that would block incoming blows (i.e. aluminum bat)

**Comment by chaosvolt** [comment]
Interesting. Still wondering what other medieval weapons you to adjust.

**Comment by Zireael07** [comment]
@md000: +1. I recall a friend of mine comparing the katana and broadsword fighting style. If the katana slid down the other's blade, it'd cut your hand, that's why all that piercing and dodging and dancing fighting style evolved. With a broadsword, the other blade is stopped by the crossguard...

**Comment by Coolthulhu** [comment]
> Maybe for another PR, but I believe parry vs block should also be looked into.

I'd say that parry and block both need to be totally rebalanced, as they don't offer much protection, especially for unskilled characters.

> Still wondering what other medieval weapons you to adjust.

If you have a python 3 environment set up and basic (totally basic, nothing out of a google tutorial) python skills, I could upload my "weapon evaluation script" to pastebin and then you could check how do they rate.

**Comment by chaosvolt** [comment]
Hmm. I'd mostly want to see what should be done to the pike, as that's about the only mainline medieval spear.

Being able to look at the script might be interesting, but it be simpler (if more time-consuming) for me to calculate rebalancing for medieval mod weapons via ye olde math. ;A;

**Comment by md000** [comment]
I have python 3 and "totally basic" is almost exactly how I would rate my python skills, but it will have to wait for tomorrow as it's 22:26 where I live and I've still got quite a bit of work to do before I go to sleep.

**Comment by Coolthulhu** [comment]
http://pastebin.com/6fp5bEHh

Just edit one line (DDA directory path) and run in python.

**Comment by chaosvolt** [comment]
Ah. More code for me to break. <3

**Comment by Coolthulhu** [comment]
Anything wrong with this PR?
While certainly not perfect, I'm pretty sure this is better than what we have now.

I documented the "formula" for balancing in the appropriate file.

**Comment by chaosvolt** [comment]
Seems alright to me. Still was pondering whether the pike needs adjusting though. :V

**Comment by Rivet-the-Zombie** [comment]
Sorry for not getting to this one sooner. I'll check this one out for merge tonight.

**Pull request merged**